lib/pigeon/dispatcher_worker.ex

```diff
@@ -46,7 +68,18 @@ defmodule Pigeon.DispatcherWorker do
 
   @impl GenServer
   def handle_call(:info, _from, %{adapter: adapter, state: state}) do
-    info = %{peername: peername(state)}
+    average_response_time_ms =
+      System.convert_time_unit(
+        state.timing_data.average_native,
+        :native,
+        :millisecond
+      )
+
+    info = %{
+      peername: peername(state),
+      average_response_time_ms: average_response_time_ms
+    }
+
```

Comment:
Two potential issues with this approach.

1. I'm not sure every call to `:info` means that the request is completed, so I don't think this will time what we want to time here.
2. For FCM/APNS, requests responses aren't necessarily going to come in order/synchronously. When requests are delayed, I think stuff is queued up, but I think in the normal case, you might be timing the start and end between the start of one notification and the end of another.

Maybe I'm not understanding the exponential moving average calculation, maybe it all averages out and I'm thinking about it too much.

Reply:
#1: it doesn't mean it's complete. It's intentionally asynchronous to that. It will give me the current weighted average that I can then use to compute a timeout for this request.

#2: Good point. I need to use the `on_response` handler to correlate responses.